castai/resource_workload_scaling_policy.go

```
@@ -0,0 +1,341 @@
+		"cpu",
+		"memory",
+	) {
+		log.Printf("[INFO] scaling policy up to date")
+		return nil
+	}
+
+	client := meta.(*ProviderConfig).api
+	clusterID := d.Get(FieldClusterID).(string)
+	req := sdk.WorkloadOptimizationAPIUpdateWorkloadScalingPolicyJSONBody{
+		Name:      d.Get("name").(string),
+		ApplyType: sdk.WorkloadoptimizationV1ApplyType(d.Get("apply_type").(string)),
+		RecommendationPolicies: sdk.WorkloadoptimizationV1RecommendationPolicies{
+			ManagementOption: sdk.WorkloadoptimizationV1ManagementOption(d.Get("management_option").(string)),
+			Cpu:              toResourcePolicies(d.Get("cpu").([]interface{})[0].(map[string]interface{})),
+			Memory:           toResourcePolicies(d.Get("memory").([]interface{})[0].(map[string]interface{})),
+		},
+	}
+
+	resp, err := client.WorkloadOptimizationAPIUpdateWorkloadScalingPolicyWithResponse(ctx, clusterID, d.Id(), req)
+	if checkErr := sdk.CheckOKResponse(resp, err); checkErr != nil {
+		return diag.FromErr(checkErr)
+	}
+
+	return resourceWorkloadScalingPolicyRead(ctx, d, meta)
+}
+
+func resourceWorkloadScalingPolicyDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*ProviderConfig).api
+	clusterID := d.Get(FieldClusterID).(string)
+
+	resp, err := client.WorkloadOptimizationAPIGetWorkloadScalingPolicyWithResponse(ctx, clusterID, d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	if resp.StatusCode() == http.StatusNotFound {
+		log.Printf("[DEBUG] Scaling policy (%s) not found, skipping delete", d.Id())
+		return nil
+	}
+	if err := sdk.StatusOk(resp); err != nil {
+		return diag.FromErr(err)
+	}
+
+	delResp, err := client.WorkloadOptimizationAPIDeleteWorkloadScalingPolicyWithResponse(ctx, clusterID, d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	if delResp.StatusCode() == http.StatusBadRequest {
+		log.Printf("[WARN] Scaling policy has active workloads (%s) and can't be deleted, removing from state", d.Id())
+		return nil
```

Shouldn't we return the error here too? Or maybe at least return Diagnostic{Severity: Warning, Summary:...}